src/unitxt/artifact.py

```diff
@@ -404,13 +407,27 @@ def __str__(self):
         return f"Artifact {self.name} does not exist, in artifactories:{self.artifactories}"
 
 
-def fetch_artifact(name):
-    if Artifact.is_artifact_file(name):
-        return Artifact.load(name), None
+def fetch_artifact(artifact_rep):
+    if isinstance(artifact_rep, Artifact):
+        return artifact_rep, None
+    if Artifact.is_artifact_file(artifact_rep):
+        return Artifact.load(artifact_rep), None
 
-    artifactory, name, args = get_artifactory_name_and_args(name=name)
-
-    return artifactory.get_with_overwrite(name, overwrite_args=args), artifactory
+    try:
```

Comment:
Instead of try catch, use the method that checks legal catalog artifact name.

Reply:
Take 413-430 become Artifact.deserialize(str)

Reply:
Done